### PR TITLE
Use existing implementation class uid and version

### PR DESCRIPTION
**Pull request by bcarthic**
#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
This PR include changes to use existing implementation Class UID and implementation Version name on DicomFile.Save and DicomFile.SaveAsync

Currenly fo-dicom updates the uid and version name to different value compared to the original file values. To keep the original values, this change is proposed

### CODE REVIEW

File: Tests/FO-DICOM.Tests/DicomFileMetaInformationTest.cs [hunk 1]
```diff
@@ -40,6 +40,78 @@ public void ImplementationVersionName_GetterWhenAttributeMissing_ReturnsNull()
...
+        {
+            var inFile = DicomFile.Open(TestData.Resolve("CT-MONO2-16-ankle"));
+            var metaInfo =
+                new DicomFileMetaInformation(
+                    inFile.FileMetaInfo,
+                    useExistingImplementionClassIdAnVersion: true);
+
+            var exception = Record.Exception(() =>
+            {
+                Assert.NotNull(metaInfo.ImplementationVersionName);
```
Comment (gofal) [inline Tests/FO-DICOM.Tests/DicomFileMetaInformationTest.cs:90]: shouldn't this asset the existence of ImplementationClassUID?

**Comment by gofal** [comment]
This option is not DICOM conformant. The intention of the two fields are, to document the creator of the binary data. You should NOT copy the previous values.
See here this table in DICOM-standard https://dicom.nema.org/medical/dicom/current/output/html/part10.html#table_7.1-1
> <html>
<body>


Implementation Class UID | (0002,0012) | 1 | Uniquely identifies the implementation that wrote this file and its content.
-- | -- | -- | --



</body>
</html>

See also here some comments on this topic in discussion groups:

https://groups.google.com/g/comp.protocols.dicom/c/zmgXL556dM0/m/59dC3g4EBeYJ
https://groups.google.com/g/comp.protocols.dicom/c/gGFooa7b7xE/m/2a6gTBRiBgAJ
https://groups.google.com/g/comp.protocols.dicom/c/piPWFxp20oE/m/t989mXA4DwAJ

even the famous dcmtk does not allow copying the value :-)
https://forum.dcmtk.org/viewtopic.php?t=4404

**Comment by gofal** [comment]
see my comment on the related issue #1544

**Comment by gofal** [comment]
please tell about the need, why you created this PR. Maybe there could be an other solution for your problem.
And if you really for some reason need this solution, then it should be more hidden or harder to implement, then just setting an option property ;-)

**Comment by bcarthic** [comment]
> please tell about the need, why you created this PR. Maybe there could be an other solution for your problem. And if you really for some reason need this solution, then it should be more hidden or harder to implement, then just setting an option property ;-)

Thanks for the review. The reason we want to keep the same value is we are working on updating a DICOM file and we need to preserve the value that is being injected by the user. I read your issue and it makes sense. I will close this PR.

**Pull request closed without merging**